python/cuml/metrics/cluster/homogeneity_score.pyx

```diff
@@ -0,0 +1,98 @@
+#
+# Copyright (c) 2019, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# cython: profile=False
+# distutils: language = c++
+# cython: embedsignature = True
+# cython: language_level = 3
+
+import numpy as np
+import cupy as cp
+
+from libc.stdint cimport uintptr_t
+
+from cuml.common.handle cimport cumlHandle
+from cuml.utils import input_to_dev_array
+import cuml.common.handle
+cimport cuml.common.cuda
+
+cdef extern from "cuml/metrics/metrics.hpp" namespace "ML::Metrics":
+    double homogeneityScore(const cumlHandle & handle, const int *y,
+                            const int *y_hat, const int n,
+                            const int lower_class_range,
+                            const int upper_class_range) except +
+
+
+def homogeneity_score(labels_true, labels_pred, handle=None):
+    """
+    Computes the homogeneity metric of a cluster labeling given a ground truth.
+
+    A clustering result satisfies homogeneity if all of its clusters contain
+    only data points which are members of a single class.
+
+    This metric is independent of the absolute values of the labels:
+    a permutation of the class or cluster label values won’t change the score
+    value in any way.
+
+    The labels in labels_pred and labels_true are assumed to be drawn from a
+    contiguous set (Ex: drawn from { 2, 3, 4}, but not from { 2, 4 }). If your
+    set of labels looks like { 2, 4 }, convert them to something like { 0, 1 }.
+
+    Parameters
+    ----------
+    handle : cuml.Handle
+    labels_pred : int32 NumPy ndarray, int32 Numba device or int32 cudf Series
+       The labels predicted by the model for the test dataset
+    labels_true : int32 NumPy ndarray, int32 Numba device or int32 cudf Series
+       The ground truth labels of the test dataset
+
+    Returns
+    -------
+    float
+      The homogeneity of the predicted labeling given the ground truth.
+      Score between 0.0 and 1.0. 1.0 stands for perfectly homogeneous labeling.
+    """
+    handle = cuml.common.handle.Handle() if handle is None else handle
+    cdef cumlHandle*handle_ = <cumlHandle*> <size_t> handle.getHandle()
+
+    cdef uintptr_t preds_ptr
+    cdef uintptr_t ground_truth_ptr
+
+    preds_m, preds_ptr, n_rows, _, _ = input_to_dev_array(
+        labels_pred,
+        convert_to_dtype=np.int32,
```

Comment:
`convert_dtype` should be a parameter of the function, and it should happen only if the user enables it, just like the other models. This type of usage can be seen in many estimators' predict: 
https://github.com/rapidsai/cuml/blob/2cea14dbcc5ff1c065d24ba537f95d45279b67a7/python/cuml/cluster/kmeans.pyx#L438

Reply:
Good point. In that case I think I should have put `check_dtype` instead of `convert_to_dtype` in the first place since I want the inputs to be `np.int32`. That's what I did on the next commit, please tell me if you disagree.